dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/schemas/SchemaGenerator.java

```diff
@@ -156,6 +159,8 @@ public static class SchemaGenerationContext {
         private Set<String> namesInUse = new HashSet<>();
         /** Map of types and their corresponding JSON schemas which should be added to defs */
         private HashMap<Type, JsonObject> defsBuilder = new HashMap<>();
+        // Maps generic type variables to concrete Types
+        private HashMap<TypeVariable<?>, Type> genericMap = new HashMap<>();
```

Comment:
I think this works because we always put things in the map before we try to look them up, but it would be better if we had a stack of maps, so we could remove mappings.

With a single map there's a risk that we hit a case where variables can't be resolved but we still have an old mapping for that variable in the map.